src/script/main.ts

```diff
@@ -1,10 +1,17 @@
+import { stat } from "fs";
+
 export class Main {
     public static viPath;
+    public static viModel;
 
     public static LoadFile(filePath: string) {
-        const currentState = history.state;
+        const currentState = top.history.state;
         const search = "?q=" + filePath;
         if (currentState !== null) {
+
+            const stateObj = {hashValue: currentState.filePath.split("#")[1], sourceModel: currentState.parsedContent};
```

Comment:
What about the null check for the #value? Doesn't this give "undefined" for PolyVIs and LVClasses?